tensorflow/python/ops/special_math_ops.py

```diff
@@ -318,44 +318,28 @@ def sort_key(input_index, a):
     # into a single axis, and combine multiple summed axes into a
     # single axis.
 
-    t0_shape = tuple(x.value for x in t0.get_shape())
+    t0_shape = _get_shape(t0)
     num_broadcast_elements_t0 = _total_size(
-        t0_shape[len(preserved_axes):-len(axes_to_sum)])
-    num_summed_elements = _total_size(t0_shape[-len(axes_to_sum):])
-    new_shape = t0_shape[:len(preserved_axes)] + (num_broadcast_elements_t0,
-                                                  num_summed_elements)
+      t0_shape[len(preserved_axes):-len(axes_to_sum)] + [1])
```

Comment:
Minor: is adding [1] necessary here?  _total_size returns 1 if given an empty list.  If for some reason this is still necessary, I think it would be cleaner to do result *= 1 at the end of _total_size.

Same question on line 331.